Zend/zend_fibers.h

```diff
@@ -0,0 +1,129 @@
+/*
+   +----------------------------------------------------------------------+
+   | Zend Engine                                                          |
+   +----------------------------------------------------------------------+
+   | Copyright (c) Zend Technologies Ltd. (http://www.zend.com)           |
+   +----------------------------------------------------------------------+
+   | This source file is subject to version 2.00 of the Zend license,     |
+   | that is bundled with this package in the file LICENSE, and is        |
+   | available through the world-wide-web at the following url:           |
+   | http://www.zend.com/license/2_00.txt.                                |
+   | If you did not receive a copy of the Zend license and are unable to  |
+   | obtain it through the world-wide-web, please send a note to          |
+   | license@zend.com so we can mail you a copy immediately.              |
+   +----------------------------------------------------------------------+
+   | Authors: Aaron Piotrowski <aaron@trowski.com>                        |
+   |          Martin Schröder <m.schroeder2007@gmail.com>                 |
+   +----------------------------------------------------------------------+
+*/
+
+#include "zend_API.h"
+#include "zend_types.h"
+
+#ifndef ZEND_FIBERS_H
+#define ZEND_FIBERS_H
+
+BEGIN_EXTERN_C()
+
+void zend_register_fiber_ce(void);
+void zend_fiber_init(void);
+void zend_fiber_cleanup(void);
+void zend_fiber_shutdown(void);
+
+extern ZEND_API zend_class_entry *zend_ce_fiber;
+
+typedef struct _zend_fiber_context zend_fiber_context;
+
+typedef void (*zend_fiber_coroutine)(zend_fiber_context *context);
+
+typedef struct _zend_fiber_stack {
+	void *pointer;
+	size_t size;
+
+#ifdef HAVE_VALGRIND
+	int valgrind;
+#endif
+} zend_fiber_stack;
+
+typedef struct _zend_fiber_context {
+	void *self;
+	void *caller;
+	zend_fiber_coroutine function;
+	zend_fiber_stack stack;
+} zend_fiber_context;
+
+#if _POSIX_MAPPED_FILES
+# include <sys/mman.h>
+# include <limits.h>
+# define ZEND_FIBER_PAGESIZE sysconf(_SC_PAGESIZE)
```

Comment:
Not a fan of calls inside "constants".

Reply:
Changed in 0123a94ba1f8378293b5a20f7ecd8b564208f816, what do you think?